diff --git a/AVOutput/AVOutputTV.h b/AVOutput/AVOutputTV.h
index 46fb07a..5157a65 100644
--- a/AVOutput/AVOutputTV.h
+++ b/AVOutput/AVOutputTV.h
@@ -248,7 +248,7 @@ class AVOutputTV : public AVOutputBase {
 		DECLARE_JSON_RPC_METHOD(getZoomModeCapsV2)
 		DECLARE_JSON_RPC_METHOD(getCMSCapsV2)
 		DECLARE_JSON_RPC_METHOD(get2PointWBCapsV2)
-		DECLARE_JSON_RPC_METHOD(getDVCalibrationCaps)
+		DECLARE_JSON_RPC_METHOD(getDolbyVisionCalibrationCaps)
 		DECLARE_JSON_RPC_METHOD(getPictureModeCapsV2)
 		DECLARE_JSON_RPC_METHOD(getAutoBacklightModeCapsV2)
 		DECLARE_JSON_RPC_METHOD(getLocalContrastEnhancementCaps)
