diff --git a/CMakeLists.txt b/CMakeLists.txt
index f73a64a3..0717375f 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -579,7 +579,7 @@ install (TARGETS subtec
 )
 
 if (CMAKE_GST_SUBTEC_ENABLED)
-    set(CMAKE_AAMP_SUBTITLE_SUPPORT TRUE)
+	set(CMAKE_AAMP_SUBTITLE_SUPPORT FALSE)
     message("CMAKE_GST_SUBTEC_ENABLED set")
     set(LIBAAMP_DEFINES "${LIBAAMP_DEFINES} -DGST_SUBTEC_ENABLED")
 endif()
