diff --git a/src/ipp2/CMakeLists.txt b/src/ipp2/CMakeLists.txt
index 05cc1a9..9931cc0 100644
--- a/src/ipp2/CMakeLists.txt
+++ b/src/ipp2/CMakeLists.txt
@@ -178,6 +178,7 @@ include_directories(${JSONCPP_INCLUDE_DIRS})
 find_path(ASIO_PATH asio.hpp)
 if(ASIO_PATH)
     include_directories(${ASIO_PATH})
+    add_definitions(-DASIO_STANDALONE)
 else(ASIO_PATH)
     message(FATAL_ERROR "Standalone Asio not found")
 endif(ASIO_PATH)
