--- a/open_cdm.h
+++ b/open_cdm.h
@@ -124,7 +124,7 @@ typedef enum
     MediaType_Video,
     MediaType_Audio,
     MediaType_Data
-} MediaType;
+} OcdmMediaType; // Renamed from "MediaType" to avoid compile error, as AAMP also has a MediaType enum

 //CENC3.0 pattern is a number of encrypted blocks followed a number of clear blocks after which the pattern repeats.
 typedef struct {
@@ -152,7 +152,7 @@ typedef struct {
 typedef struct {
     uint16_t height;
     uint16_t width;
-    MediaType media_type;
+    OcdmMediaType media_type;
 } MediaProperties;
